docs/reference/components/sinks/gcp_pubsub.cue

```diff
@@ -83,6 +83,16 @@ components: sinks: gcp_pubsub: {
 				examples: ["/path/to/credentials.json"]
 			}
 		}
+		endpoint: {
```

Comment:
I think we should document `skip_authentication` here, even though it's only really used for self tests.

Reply:
is there a concept of "private parameters", ie undocumented ones that are used for testing.  For example I just went and changed `emulator_host` to `endpoint` without any regard to who might be using it because I assumed it would only be used internally.  If we "publicise" `skip_authentication`, do we then need to be careful about deprecating it in the future?  I can imagine that this may one day evolve to reflect a bunch of possible auth options and not just the binary on/off

Reply:
We do have a concept of undocumented options and sources. So yes, if an option is not explicitly listed in our `.cue` files then it can be changed. What's in those files is our public contract.

Reply:
Ah perfect.  Then might I suggest this remain undocumented to retain some flexibility, until some public use case emerges.  Its existence currently is because it keeps the code cleaner and more expressive and less coupled to the hostnames used for testing.